src/cares_wrap.cc

```diff
@@ -1797,22 +1797,24 @@ static void Query(const FunctionCallbackInfo<Value>& args) {
 
   Local<Object> req_wrap_obj = args[0].As<Object>();
   Local<String> string = args[1].As<String>();
-  Wrap* wrap = new Wrap(channel, req_wrap_obj);
+  std::unique_ptr<Wrap> wrap {new Wrap(channel, req_wrap_obj)};
 
   node::Utf8Value name(env->isolate(), string);
   channel->ModifyActivityQueryCount(1);
   int err = wrap->Send(*name);
   if (err) {
     channel->ModifyActivityQueryCount(-1);
-    delete wrap;
+  } else {
+    wrap.release();
```

Comment:
Could you add a comment why this is needed.

Reply:
I can add a comment for sure though I've not done so in previous cases where this pattern has been used. I'll wrap it with `USE`. Thanks